### PR TITLE
Add notification settings model and API

### LINKED ISSUE #459: Model and API for tracking users' digest email preferences

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #459 

#### What's this PR do?
Adds settings API for email nottifications

#### How should this be manually tested?

- Go to `/api/v0/notification_settings/`, verify you get an empty list
- Run `./manage.py populate_notification_settings --username USERNAME`, verify you get a `NotificationSetting` record for the frontpage
- Go to `/api/v0/notification_settings/`, verify you get that same setting in the list
- Go to `/api/v0/notification_settings/frontpage/`, verify you get that same setting
- Verify you can patch the setting with a new `trigger_frequency` value

### CODE REVIEW

File: notifications/models.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from django.conf import settings
+from django.db import models
+
+from open_discussions.models import TimestampedModel
+
+NOTIFICATION_TYPE_FRONTPAGE = "frontpage"
+NOTIFICATION_TYPE_COMMENTS = "comments"
+NOTIFICATION_TYPE_CHOICES = (
+    (NOTIFICATION_TYPE_FRONTPAGE, "Frontpage"),
+    (NOTIFICATION_TYPE_COMMENTS, "Coments"),
```
Comment (aliceriot) [inline notifications/models.py]: little typo (`Coments`)

**Comment by aliceriot** [comment]
It seems the url for the API is actually `/api/v0/notification_settings_api/`, is that what we want?

**Comment by aliceriot** [comment]
functionality looks good otherwise though!

File: notifications/models.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+from django.conf import settings
+from django.db import models
+
+from open_discussions.models import TimestampedModel
+
+NOTIFICATION_TYPE_FRONTPAGE = "frontpage"
+NOTIFICATION_TYPE_COMMENTS = "comments"
+NOTIFICATION_TYPE_CHOICES = (
+    (NOTIFICATION_TYPE_FRONTPAGE, "Frontpage"),
+    (NOTIFICATION_TYPE_COMMENTS, "Comments"),
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply notifications/models.py]: Yeah, they show up as a label in certain places in general (e.g. django-admin).

**Comment by rhysyngsun** [comment]
@aliceriot fixed the api url

**Pull request merged**